src/index.d.ts

```diff
@@ -191,34 +191,17 @@ declare namespace preact {
 	 * Global options for preact
 	 */
 	interface Options {
-		/** Attach a hook that is invoked before render, mainly to check the arguments. */
-		root?(vnode: ComponentChild, parent: Element | Document | ShadowRoot | DocumentFragment): void;
 		/** Attach a hook that is invoked whenever a VNode is created. */
-		vnode(vnode: VNode): void;
-		/** Attach a hook that is invoked after a tree was mounted or was updated. */
-		commit?(vnode: VNode): void;
-		/** Attach a hook that is invoked immediately before a component is unmounted. */
+		vnode?(vnode: VNode): void;
+		/** Attach a hook that is invoked immediately before a vnode is unmounted. */
 		unmount?(vnode: VNode): void;
 		/** Attach a hook that is invoked before a vnode is diffed. */
 		diff?(vnode: VNode): void;
-		/** Attach a hook that is invoked before a vnode has rendered. */
-		render?(vnode: VNode): void;
```

Comment:
Could this be accomplished with a `vnode` hook that wraps a component's `render` method, similar to the wrapping I referred to in `root`? Wrapping the `render` method would probably be way more useful cuz that way the wrapper gets access to the component (through `this`) and the `vnode` (as the parameter to the `vnode` hook). Currently a public `render` option wouldn't be able to access the component of the VNode cuz it's a private property on the VNode.

Reply:
I like this idea, never thought about it that way :+1: That would be a lot more powerful and would likely be easier to understand for users :+1: